### ISSUE TITLE
AQL UPSERT simplification when UPDATE / REPLACE is {}

### ISSUE THREAD

**Issue by baslr** [arangodb/arangodb#2013]
```
could this AQL query

UPSERT {_key: 'myKey'}
INSERT {_key: 'myKey', abc: 123}
UPDATE {} IN myCol

be simplified to

UPSERT {_key: 'myKey'}
INSERT {_key: 'myKey', abc: 123} IN myCol
```

**Comment by jsteemann** [comment]
This would cause conflicts in the parser grammar.
The rule for UPSERT currently is
```
update_or_replace:
    T_UPDATE { ... }
  | T_REPLACE { ... }
;

upsert_statement:
  T_UPSERT  expression  T_INSERT  expression  update_or_replace  expression  in_or_into_collection  options  { ... }
```

Turning it into
```
optional_update_or_replace_with_expression:
    /* empty */ { ... }
  | T_UPDATE  expression { ... }                         
  | T_REPLACE  expression { ... }
;

upsert_statement:
  T_UPSERT  expression  T_INSERT  expression  optional_update_or_replace_with_expression  in_or_into_collection  options { ... }
```
leads to
```
Aql/grammar.y: error: 3 shift/reduce conflicts [-Werror=conflicts-sr]
```

**Comment by baslr** [comment]
so its currently not possible to assume `update {}` as default update strategy?

**Comment by jsteemann** [comment]
Currently it's not possible.
It would be trivial to add the default behavior to the UPSERT operation code, but it's not trivial to allow this syntax in the AQL parser AFAIK.

**Comment by dothebart** [comment]
@baslr since it doesn't seem to be possible to get a working grammar definition with bison do you have other suggestions how we could achieve this? Else I don't see any other chance than closing this as immpossible.

**Comment by pekeler** [comment]
You could add something like `getsert` or `optsert` :)
Just yesterday I also used `upsert` with empty update.

**Comment by Simran-B** [comment]
The explicit `UPDATE {}` is something one can live with pretty good IMO.

BTW: A feature that was indirectly requested on Slack today was kind of an UPMOVE (update or remove). There would also be cases in which some people would want to conditionally REMOVE I guess, like `UPSERT {_key: "1234"} UPDATE {} REMOVE {_key: "1234"} IN coll`.

**Comment by jsteemann** [comment]
@Simran-B : isn't the latter just an remove that should not fail if the document does not exist, e.g.
`REMOVE { _key: "1234" } IN coll OPTIONS { ignoreErrors: true }`?

**Comment by Simran-B** [comment]
True... the actual use case was to either UPDATE or REMOVE though. I just figured that you might reconsider and just make the UPDATE blank but keep the UPSERT / UPMOVE construct. Call it syntactic sugar.
